transforms/utils/replaceType.js

```diff
@@ -113,8 +113,16 @@ function replaceReactType(
 		},
 	);
 	for (const typeReferences of sourceIdentifierTypeReferences) {
-		const changedIdentifiers = typeReferences.replaceWith((path) => {
-			return buildTargetTypeReference(path.value);
+		const changedIdentifiers = typeReferences.forEach((path) => {
+			const targetNode = buildTargetTypeReference(path.value);
+			if (
+				targetNode.type === "TSUnionType" &&
+				path.parentPath.value.type === "TSArrayType"
+			) {
+				path.replace(j.tsParenthesizedType(targetNode));
```

Comment:
I don't trust this. Can we just directly replace it with a newly created array type. Something like 
```suggestion
				path.replace(j.tsArrayType(targetNode));
```
?

Reply:
Thanks for the review! I don't think that specific suggestion would work at it would result in: 

```diff
- ReactChild[]
+ ReactElement | number | string[][]
```

I added a test for the `Array<>` generic and pushed [a PR](https://github.com/benjamn/recast/pull/1416) to `recast` so this is on their radar.

Reply:
Thank you for digging!